# Assignment 1
## Dataset: employee.csv

In [1]:
# Setup
import pandas as pd

# Import file as dataframe
df = pd.read_csv('http://myy.haaga-helia.fi/~menetelmat/Data-analytiikka/Teaching/employee.csv')

# And take a sneak peek of what's inside
print('Rows: ' + str(df.shape[0]))
print('Cols: ' + str(df.shape[1]))
df.head()


Rows: 1470
Cols: 35


Age Attrition     BusinessTravel  DailyRate              Department  \
0   41       Yes      Travel_Rarely       1102                   Sales   
1   49        No  Travel_Frequently        279  Research & Development   
2   37       Yes      Travel_Rarely       1373  Research & Development   
3   33        No  Travel_Frequently       1392  Research & Development   
4   27        No      Travel_Rarely        591  Research & Development   

   DistanceFromHome  Education EducationField  EmployeeCount  EmployeeNumber  \
0                 1          2  Life Sciences              1               1   
1                 8          1  Life Sciences              1               2   
2                 2          2          Other              1               4   
3                 3          4  Life Sciences              1               5   
4                 2          1        Medical              1               7   

   ...  RelationshipSatisfaction StandardHours  StockOptionLevel  \
0  ...                         1            80                 0   
1  ...                         4            80                 1   
2  ...                         2            80                 0   
3  ...                         3            80                 0   
4  ...                         4            80                 1   

   TotalWorkingYears  TrainingTimesLastYear WorkLifeBalance  YearsAtCompany  \
0                  8                      0               1               6   
1                 10                      3               3              10   
2                  7                      3               3               0   
3                  8                      3               3               8   
4                  6                      3               3               2   

  YearsInCurrentRole  YearsSinceLastPromotion  YearsWithCurrManager  
0                  4                        0                     5  
1                  7                        1                     7  
2                  0                        0                     0  
3                  7                        3                     0  
4                  2                        2                     2  

[5 rows x 35 columns]

### Describing the continuous variables
Let's start by looking what's happening with the numbers.
Here transposing the output provides a nicer view

In [7]:
df.describe().transpose()

NameError: name 'np' is not defined

### Visualising distributions

In [5]:
import matplotlib

## Initial findings on numeric variables
* Based on count of 1470, all variables were captured for each of the employees in the sample.

In [14]:
# Now let's see what's up with the categorical ones
df.describe(include='object').transpose()

count unique                     top  freq
Attrition       1470      2                      No  1233
BusinessTravel  1470      3           Travel_Rarely  1043
Department      1470      3  Research & Development   961
EducationField  1470      6           Life Sciences   606
Gender          1470      2                    Male   882
JobRole         1470      9         Sales Executive   326
MaritalStatus   1470      3                 Married   673
Over18          1470      1                       Y  1470
OverTime        1470      2                      No  1054

In [150]:

#cats = df.select_dtypes(include='object').value_counts()
#print(cats)

#print(df.value_counts)
#print(df.select_dtypes(include='object').keys())


#.apply(lambda x: x.value_counts(), result_type='expand')

#dfcat = []
#row = pd.Series()
for i in df.select_dtypes(include='object').columns.tolist():
     levels = df[i].value_counts()
     levels.name=i
     print(levels.to_frame())
#print(catVars)
#df.JobRole.value_counts()

     Attrition
No        1233
Yes        237
                   BusinessTravel
Travel_Rarely                1043
Travel_Frequently             277
Non-Travel                    150
                        Department
Research & Development         961
Sales                          446
Human Resources                 63
                  EducationField
Life Sciences                606
Medical                      464
Marketing                    159
Technical Degree             132
Other                         82
Human Resources               27
        Gender
Male       882
Female     588
                           JobRole
Sales Executive                326
Research Scientist             292
Laboratory Technician          259
Manufacturing Director         145
Healthcare Representative      131
Manager                        102
Sales Representative            83
Research Director               80
Human Resources                 52
          MaritalStatus
Married             673
Singl